docs/api/types.md

```diff
@@ -160,8 +160,8 @@ as [Resource](./resource) expects it
 
 ```typescript
 export type RestFetch<
-  P = object,
+  P = any,
```

Comment:
why any?

Reply:
This was already discussed when it first landed. Due to typescripts limitations this is the only way to allow users to specify their own types. The alternative would be not allowing for precise typing. You started a whole issue on typescript  repo because of this

Reply:
https://github.com/coinbase/rest-hooks/pull/479

Again, this is unfortunately one of the things where Flow actually got it better than TypeScript.

Stated goals are a [Resource definition works without defining endpoints](https://resthooks.io/docs/getting-started/usage):

```typescript
import { Resource } from '@rest-hooks/rest';

export default class ArticleResource extends Resource {
  readonly id: number | undefined = undefined;
  readonly title: string = '';
  readonly content: string = '';
  readonly author: number | null = null;
  readonly tags: string[] = [];

  pk() {
    return this.id?.toString();
  }

  static urlRoot = 'http://test.com/article/';
}
```

```typescript
ArticleResource.list()
ArticleResource.detail()
//....
```

And secondly, [overrides with precise types](https://resthooks.io/docs/guides/rest-types) are possible:

```typescript
class MyResource extends Resource {
  static detail<T extends typeof Resource>(
    this: T,
  ): RestEndpoint<RestFetch<{ id: string }>, T, undefined> {
    return super.list();
  }
}
```

Reply:
To clarify, this line is only a docs update to correctly match the types as I missed that in the previous PR